tls/s2n_server_new_session_ticket.c

```diff
@@ -184,7 +184,8 @@ static S2N_RESULT s2n_generate_ticket_lifetime(struct s2n_connection *conn, uint
     RESULT_ENSURE_REF(conn);
     RESULT_ENSURE_MUT(ticket_lifetime);
 
-    uint32_t key_lifetime_in_secs = conn->config->decrypt_key_lifetime_in_nanos / ONE_SEC_IN_NANOS;
+    uint32_t key_lifetime_in_secs =
+            (conn->config->encrypt_decrypt_key_lifetime_in_nanos + conn->config->decrypt_key_lifetime_in_nanos) / ONE_SEC_IN_NANOS;
```

Comment:
I remember having a conversation about why we should only consider the decrypt_key_lifetime here. However, I don't remember the rational. What is more correct about this change?

Reply:
The full key lifetime is the combination of the two. You can see it in the TLS1.2 logic: https://github.com/aws/s2n-tls/blob/main/tls/s2n_server_new_session_ticket.c#L83 We would only ignore the "encrypt_decrypt" lifetime if it was included in the "decrypt" lifetime, but the TLS1.2 logic makes it clear that's not the case.